src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -43,12 +51,30 @@ private void handleCommandEntered() {
 
         try {
             commandExecutor.execute(commandText);
-            commandTextField.setText("");
+            commandHistory.addCommand(commandText, commandTextField);
```

Comment:
Does this only add commands into the history if it was a successful execution?

Reply:
It would be more intuitive to add any command that has been executed into history

Reply:
> Does this only add commands into the history if it was a successful execution?

Yes, since `commandExecutor.execute(commandText)` would throw an exception if it is not a valid command



> It would be more intuitive to add any command that has been executed into history

Currently if a command is not executed correctly, the code in the catch block is executed, which highlights the incorrect command in the text field. If we do this, if the user presses UP after executing an incorrect command, they would see the same incorrect command in history, which doesn't really make sense

Reply:
It would make sense imo. 
When you enter a wrong command, lets say you missed out a space, you would want to reuse your previous wrong input, and just add the space in. This is how its implemented in terminal on macos

Reply:
As discussed, currently typing a wrong command would cause it to be highlighted red in the text field, instead of emptying the text field like in terminal, allowing the user to edit the wrong command. Thus we should not add it into the history as there is no extra added value in this.